### PR TITLE
Feature/request presentations

### PR DESCRIPTION
**Pull request by jacquesporveau**
Part two of https://www.pivotaltracker.com/story/show/140356337

### CODE REVIEW

File: app/controllers/topic_requests_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  def index
+    @topic_requests = TopicRequest.all
+    @topic_request = TopicRequest.new
+  end
+
+  def create
+    topic_request = TopicRequest.new(topic_request_params)
+    authorize topic_request
+
+    flash[:notice] = if topic_request.save
```
Comment (graygilmore) [inline app/controllers/topic_requests_controller.rb]: Should we do a `flash[:success]` and `flash[:error]` instead?

File: app/controllers/topic_requests_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                       'Topic request was succesfully created.'
+                     else
+                       'Error in creating topic request.'
+                     end
+    redirect_to requests_presentations_path
```
Comment (graygilmore) [inline app/controllers/topic_requests_controller.rb]: If we do a redirect here do we get specific error messages for why it wasn't submitted? Like if the user leaves it blank.

File: app/policies/topic_request_policy.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class TopicRequestPolicy < ApplicationPolicy
+  def create?
+    true
```
Comment (graygilmore) [inline app/policies/topic_request_policy.rb]: Should we skip this entirely and just not do `authorize topic_request`?

File: spec/policies/topic_request_policy_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe TopicRequestPolicy do
+  let(:topic_request) { FactoryGirl.build(:topic_request) }
+  context 'the person is not signed in' do
+    subject { described_class.new(nil_person, topic_request) }
+    let(:nil_person) { nil }
+
+    describe '#create?' do
```
Comment (graygilmore) [inline spec/policies/topic_request_policy_spec.rb]: I think this spec file is a little backwards. Normally you'd see something like:

```ruby
describe "#my_method" do
  subject { my_method }

  it "does something" do
  end

  context "context one" do
    it "does another thing" do
    end
  end

  context "context two" do
    it "does something else" do
    end
  end
end
```

**Review by graygilmore — CHANGES_REQUESTED** [review]

File: app/controllers/topic_requests_controller.rb (hunk 2 above)
Comment (jacquesporveau) [reply app/controllers/topic_requests_controller.rb]: Yes it did do that.

File: app/policies/topic_request_policy.rb (hunk 3 above)
Comment (jacquesporveau) [reply app/policies/topic_request_policy.rb]: Yes we should. Done now.

File: spec/policies/topic_request_policy_spec.rb (hunk 4 above)
Comment (jacquesporveau) [reply spec/policies/topic_request_policy_spec.rb]: Point taken. This code has all been removed along with the policy it was testing though.

**Review by natyork — COMMENTED** [review]
Looks good to me :) Nice work!

File: app/controllers/topic_requests_controller.rb (hunk 1 above)
Comment (jacquesporveau) [reply app/controllers/topic_requests_controller.rb]: :+1:

**Review by graygilmore — APPROVED** [review]

**Pull request merged**